### PR TITLE
Nn 2718 csra details page

### PR DESCRIPTION
**Pull request by Nimphal**
Screenshot from 2020-08-27 15-14-12

### CODE REVIEW

File: backend/controllers/cellMove/viewCellSharingAssessmentDetails.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+const { putLastNameFirst, hasLength } = require('../../utils')
+const { getBackLinkData } = require('./cellMoveUtils')
+
+module.exports = ({ elite2Api, logError }) => async (req, res) => {
+  const { offenderNo } = req.params
+
+  try {
+    const { firstName, lastName, assignedLivingUnit } = await elite2Api.getDetails(res.locals, offenderNo, true)
+
+    const assessments = await elite2Api.getAssessments(res.locals, { offenderNumbers: [offenderNo], code: 'CSR' })
```
Comment (Nimphal) [inline backend/controllers/cellMove/viewCellSharingAssessmentDetails.js:12]: The offender details endpoint does not return the approval date or the agency Id, so calling an assessment specific endpoint to get all the data. Can be changed to the details page in the future if we update the API.

File: backend/controllers/cellMove/viewCellSharingAssessmentDetails.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+const moment = require('moment')
+const { serviceUnavailableMessage } = require('../../common-messages')
+const { putLastNameFirst, hasLength } = require('../../utils')
+const { getBackLinkData } = require('./cellMoveUtils')
+
+module.exports = ({ elite2Api, logError }) => async (req, res) => {
+  const { offenderNo } = req.params
+
+  try {
+    const { firstName, lastName, assignedLivingUnit } = await elite2Api.getDetails(res.locals, offenderNo, true)
```
Comment (JamieDraperUK) [inline backend/controllers/cellMove/viewCellSharingAssessmentDetails.js:10]: I think the unspoken rule is to group more than one call into a Promise.all() if possible.

File: integration-tests/integration/cellMove/viewCellSharingRiskAssessmentDetails.spec.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
+const offenderFullDetails = require('../../mockApis/responses/offenderFullDetails')
```
Comment (JamieDraperUK) [inline integration-tests/integration/cellMove/viewCellSharingRiskAssessmentDetails.spec.js:1]: No fancy Page object? (I like this way better)

Comment (JamieDraperUK) [reply integration-tests/integration/cellMove/viewCellSharingRiskAssessmentDetails.spec.js:1]: Actually, sorry I'm blind. I can see it below 👁️

File: integration-tests/integration/cellMove/viewCellSharingRiskAssessmentDetails.spec.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+          assessorUser: 'TEST_USER',
+        },
+      ],
+    })
+  })
+
+  it('Shows the correct data for non-associations', () => {
+    cy.visit(`/prisoner/${offenderNo}/cell-move/cell-sharing-risk-assessment-details`)
+    const cellSharingRiskAssessmentDetailsPage = CellSharingRiskAssessmentPage.verifyOnPage()
+    cy.get('.govuk-summary-list--no-border').then($section => {
```
Comment (JamieDraperUK) [inline integration-tests/integration/cellMove/viewCellSharingRiskAssessmentDetails.spec.js:46]: Should this be encapsulated inside the page object?

File: integration-tests/integration/cellMove/viewCellSharingRiskAssessmentDetails.spec.js (hunk 3 above)
Comment (Nimphal) [reply integration-tests/integration/cellMove/viewCellSharingRiskAssessmentDetails.spec.js:1]: That's just the json response, there's a fancy page object underneath I'm afraid :D

**Review by JamieDraperUK — APPROVED** [review]
LGTM

File: integration-tests/integration/cellMove/viewCellSharingRiskAssessmentDetails.spec.js (hunk 4 above)
Comment (Nimphal) [reply integration-tests/integration/cellMove/viewCellSharingRiskAssessmentDetails.spec.js:46]: we haven't been doing that for summaries as various parts of them are optional. Less so in this one, so probably wouldn't hurt. I'll update when next in the area, I'm sure it won't be long

**Review by JamieDraperUK — APPROVED** [review]
LGTM!

**Pull request merged**